### PR TITLE
Add search type link

### LINKED ISSUE #2313: [FR] support search URLs
opening from external apps. Think URL intents should be added and such...

- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I checked if the issue/feature exists in the latest version.
- [ ] I did use the [incredible bugreport to markdown converter](https://teamnewpipe.github.io/CrashReportToMarkdown/) to paste bug reports.

### PR DESCRIPTION
**Pull request by Sola0404**
#### What is it?
- [ ] Bugfix (user facing)
- [x] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- Add SEARCH type case in the intent handler
- Add a method to generate an intent conveying the information of service id, URL, link type, and search string. This method is used in getIntentByLink when link type is SEARCH, distinguishing the conditions between using SEARCH link and open search.

#### Fixes the following issue(s)
Fixes #2313 

#### PR relies on
https://github.com/TeamNewPipe/NewPipeExtractor/pull/950

#### APK testing

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AudricV** [comment]
Could you explain better what you added in this pull request? Also, it seems you did extractor changes for this, but you didn't use them in this PR. Please do so if that's needed.

**Comment by Sola0404** [comment]
Thank you for your response. I've updated the description. 
I changed extractor for this issue, but I don't know how to include these changes in this PR, cause they are from different repositories. Could you give me some advice? Thanks in advance!

**Comment by Stypox** [comment]
You need to create a pull request in NewPipeExtractor, too. As you noticed, they are two different repositories, hence two different PRs ;-)

**Comment by ShareASmile** [comment]
> @Sola0404 You need to create a pull request in NewPipeExtractor, too.

@Stypox it's already been opened on extractor https://github.com/TeamNewPipe/NewPipeExtractor/pull/950

AudricV wrote:
> it seems you did extractor changes for this, but you didn't use them in this PR. Please do so if that's needed.

Stypox @Sola0404 is asking how to use extractor commits in this PR

File: app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java [hunk 1]
```diff
@@ -672,6 +686,11 @@ public static Intent getIntentByLink(final Context context,
                     + " url=" + url);
         }
 
+        if (linkType == StreamingService.LinkType.SEARCH) {
+            return getOpenSearchIntent(context, url, service.getServiceId(),
+                    linkType, service.getIdByUrl(url));
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java:691]: Intead of obtaining the ID this way, as said in https://github.com/TeamNewPipe/NewPipeExtractor/pull/950#discussion_r1006825163, I think you should use `service.getSearchLHFactory().getId(url)`.

**Review by Stypox — CHANGES_REQUESTED** [review]
Are urls copied to the search field in NewPipe handled as opened URLs?

**Comment by Stypox** [comment]
I'm closing this for now, both because of inactivity, and because we're going to start the rewrite soon and this would get conflicts despite its simplicity. I'm leaving https://github.com/TeamNewPipe/NewPipeExtractor/pull/950#issuecomment-2025494747 open though, so once that is done and the rewrite is in a good shape this can be reopened. Thank you!

**Pull request closed without merging**